File: Web_UI/ClientApp/src/app/models/Comercio.ts
import { Archivo } from "./Archivo";
import { Categoria } from "./categoria.model";

export class Comercio {
  id: number;
  idAdmin: number;
  nombre: string;
  nombreLegal: string;
  cedulaJuridica: string;
  archivos: Archivo[];
  categorias: string[];
  estado: number;

}

File: Entities/Comercio.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Text.Json.Serialization;

namespace Entities
{
    public class Comercio : BaseEntity
    {
        [JsonPropertyName("id")]
        public int Id { get; set; }
        [JsonPropertyName("idAdmin")]
        public int IdAdmin { get; set; }
        [JsonPropertyName("nombre")]
        public string Nombre { get; set; }
        [JsonPropertyName("nombreLegal")]
        public string NombreLegal { get; set; }
        [JsonPropertyName("cedulaJuridica")]
        public string CedulaJuridica { get; set; }
        [JsonPropertyName("archivos")]
        public Archivo[] Archivos { get; set; }
        [JsonPropertyName("categorias")]
        public string[] Categorias { get; set; }
        [JsonPropertyName("estado")]
        public int Estado { get; set; }

        public Comercio()
        {

        }
    }
}